pvlib/modelchain.py

```diff
@@ -994,7 +998,10 @@ def _set_celltemp(self, model):
                                   self.results.effective_irradiance)
         temp_air = _tuple_from_dfs(self.weather, 'temp_air')
         wind_speed = _tuple_from_dfs(self.weather, 'wind_speed')
-        self.results.cell_temperature = model(poa, temp_air, wind_speed)
+        arg_list = [poa, temp_air, wind_speed]
+        if model == self.system.noct_sam_celltemp:
+            arg_list += [self.results.effective_irradiance]
+        self.results.cell_temperature = model(*tuple(arg_list))
```

Comment:
```suggestion
        kwargs = {}
        if model == self.system.noct_sam_celltemp:
            kwargs['effective_irradiance'] = self.results.effective_irradiance
        self.results.cell_temperature = model(poa, temp_air, wind_speed, **kwargs)
```

Reply:
I like to pass keyword arguments as keyword arguments, but I don't feel strongly about it so feel free to reject.